app/service/planning_svc.py

```diff
@@ -123,3 +128,20 @@ def _reward_fact_relationship(combo_set, combo_link, score):
         if not operation['cleanup']:
             for link in links:
                 link['cleanup'] = None
+
+    async def _explode_private_facts(self, op_id, agent_id):
+        """
+        Only select the id of the agent's collected facts during the operation
+        """
+        sql = """
+        SELECT 
+            b.id 
+        FROM core_chain a 
+            JOIN (SELECT id, link_id FROM core_fact GROUP BY id) b 
+              ON a.id=b.link_id 
+        WHERE 
+            a.op_id = %s AND a.host_id = %s;
+        """ % (op_id, agent_id)
+        private_facts =  await self.data_svc.dao.raw_select(sql)
```

Comment:
I messed you up by removing the raw_select from the data_svc DAO object (in an attempt to push everyone to the data_svc instead of hitting the DAO directly, you'll notice I changed all the plugins/code to do this). you'll want to do this get in a different way-- multiple calls to the DB aren't so bad here, as by default we are using the in-memory database version of sqlite3.